src/coauthor_interface/frontend/js/async.js

```diff
@@ -1,6 +1,7 @@
 async function startSession(accessCode) {
   session = {};
   logs = [];
+  sessionEnded = false;  // Reset session ended flag for new session
```

Comment:
The use of this flag feels clunky to me, but I'm not familiar enough with the code to propose a different solution. Is there another variable that maintains session information that can be used instead? (This is something that would fall under the category of "aggressively eliminate state").